src/Tests/TransactionTest.php

```diff
@@ -419,7 +410,7 @@ public function test_additional_property_reason()
 
     public function test_additional_property_is_capturable()
```

Comment:
Functions like  test_additional_property_is_capturable, test_method_is_capturable, test_additional_property_order_id
Looks similar. I offer the following option to reduce repetitions (it should be tested):

````
public function test_additional_property_is_capturable()
{
    $real = $this->getRealForTest(["is_capturable" => true]);
    $this->assertToArrayTest(
        $real,
        true,
        "is_capturable"
    );    
}

public function test_method_is_capturable()
{
    $real = $this->getRealForTest(["is_capturable" => true]);
    $this->assertIsCapturableTest(
        $real,
        true
    );
}

public function test_additional_property_order_id()
{    
    $real = $this->getRealForTest(["order_id" => "123"]);
    $this->assertToArrayTest(
        $real,
        "123",
        "order_id"
    );   
}

public function test_additional_property_channel_valid()
{
    $real = $this->getRealForTest(["channel" => "ecom"]);
    $this->assertToArrayTest(
        $real,
        "ecom",
        "channel"
    );    
}

public function test_additional_property_project_type()
{
    $real = $this->getRealForTest(["project_type" => "ginger"]);
    $this->assertToArrayTest(
        $real,
        "ginger",
        "project_type"
    );
}

private function getRealForTest($mockedData)
{
    $real = $this->client->fromArray(
        $real = Client::fromArray(
            Transaction::class,
            array_merge(
                self::MOCK_DATA_FOR_TRANSACTION,
                $mockedData
            )
        );
   return $real;
}

private function assertToArrayTest($real, $expected, $realId)
{    
    return self::assertSame(
        $real->toArray()[$realId],
        $expected
    );
}

private function assertIsCapturableTest($real, $expected)
{
    self::assertSame(
        $real->isCapturable(),
        $expected
    );
}
```
What do you thinks about such changes?

Reply:
your suggestion is good but i think testing additional property is out of scope for now due to latest implemented method additional property filter